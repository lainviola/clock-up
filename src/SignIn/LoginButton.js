// @flow
import React, { Component, PropTypes } from 'react'
import Auth0Lock from 'auth0-lock'
import { withRouter } from 'react-router-dom'
import { AUTH0_ID_TOKEN } from '../GlobalConst'

type Props = {
  // withRouter()
  match: any,
  location: any,
  history: any
}

class LoginButton extends Component<Props> {
  lock: Auth0Lock

  constructor(props) {
    super(props)

    // TODO constructor injection & add test
    const clientId = process.env.REACT_APP_AUTH0_CLIENT_ID
    const domain = process.env.REACT_APP_AUTH0_DOMAIN
    this.lock = new Auth0Lock(clientId, domain)
  }

  componentDidMount() {
    // Auth0のログインモーダルで認証 -> コールバックURLへ帰還した時の処理
    this.lock.on('authenticated', authResult => {
      window.localStorage.setItem(AUTH0_ID_TOKEN, authResult.idToken)
      this.props.history.push(`/signup`)
    })
  }

  /**
   * Auth0のログインモーダルを表示
   */
  showLogin = () => {
    this.lock.show({
      auth: {
        params: {
          responseType: 'id_token token'
        }
      }
    })
  }

  render() {
    return (
      <div>
        <span onClick={this.showLogin} data-test="sign-in-btn">Google Login</span>
      </div>
    )
  }
}

export default withRouter(LoginButton)
