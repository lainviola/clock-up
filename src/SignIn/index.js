// @flow
import React, { Component } from 'react'
import { withRouter } from 'react-router-dom'
import LoginButton from './LoginButton'

type Props = {
  data: any,
  // withRouter()
  match: any,
  location: any,
  history: any
}

class SignIn extends Component<Props> {
  render() {
    return <LoginButton />
  }
}

export default withRouter(SignIn)
