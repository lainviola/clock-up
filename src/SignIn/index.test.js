import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import SignIn from './index'

describe('Component: SignIn', () => {
  it('snapshot', () => {
    const wrapper = shallow(<SignIn />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
