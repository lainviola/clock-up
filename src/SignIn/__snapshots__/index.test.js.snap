// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: SignIn snapshot 1`] = `
<Route
  render={[Function]}
/>
`;
